benchmarks/README.md

```diff
@@ -76,3 +83,19 @@ jlpm run start:compare
 # Create and open graph to view results
 jlpm run start:compare-graph
 ```
+
+
+When creating the comparison file, it will also print out a textual form of the analysis, like this:
+
+```bash
+$ node lib/compare.js
+Writing output to diff.csv
+Parsing data { path: 'old.csv', version: 'old' }
+Parsing data { path: 'old.csv', version: 'new_' }
+In firefox opening a notebook with 100 n outputs each of a div where n=0 is between 36.4% slower and 26.7% faster with 95% confidence.
+In firefox opening a notebook with one output with 100 n divs where n=0 is between 1615.4% slower and 94.2% faster with 95% confidence.
+...
+✨  Done in 1.08s.
+```
+
```

Comment:
Super. We could have also a json like the following written on disk - I still need to make my mind on `between 1615.4% slower and 94.2% faster`

``
{
  "firefox": {
     "lower_interval_change_percent": -36.4,
     "upper_interval_change_percent": 26.7,
     "confidence_interval_percent": 95
   },
  "firefox": {
    ...
  }
}
```

Reply:
`1615.4% slower` means that the ratio between the two performances is `1:17` basically. i..e the new result is 17 times slower than the old one.


Instead, we could phrase this same thing as `860%+- 850%`?

The real issue I guess is there is a lot of variance in that run, one big outlier, that messes it up:

```
open,firefox,0,one output with 100 n divs,149
open,firefox,0,one output with 100 n divs,138
open,firefox,0,one output with 100 n divs,145
open,firefox,0,one output with 100 n divs,193
open,firefox,0,one output with 100 n divs,183
open,firefox,0,one output with 100 n divs,1350
open,firefox,0,one output with 100 n divs,194
open,firefox,0,one output with 100 n divs,159
open,firefox,0,one output with 100 n divs,167
open,firefox,0,one output with 100 n divs,173
```

So we could figure out how to make this more sensible, but the real issue is a data collection issue. Is the `1350` an accurate measurement? Did it really take a long time once out of every 10 times? If so, then yeah it's much harder to give an accurate comparison between two benchmarks.

Reply:
> We could have also a json like the following written on disk 

That's basically what the `diff.csv` file is. The only difference is it records the mean and size of the error bounds, instead of the two sides of the bounds.

Reply:
outliers will be something we have to manage. They will come on the Desktop if you ask to browse a page on top of the running benchmark, on the CI the VM may be temporary under pressure. Does the paper talk about that and proposes solutions?

Looking at the diff.csv, it contains as many rows/datapoints as benchmark run. I was thinking more to a aggregated/summary output like the sentence you propose (need to understand a bit more to have a correct phrasing - That phrasing should be understable by someone having no knowledge of the used methodology)

Reply:
> outliers will be something we have to manage. They will come on the Desktop if you ask to browse a page on top of the running benchmark, on the CI the VM may be temporary under pressure. Does the paper talk about that and proposes solutions?

Run more trials maybe ;) Or run in a more stable environment (would require a dedicated CI box probably). 

Also see this newer paper ["Software microbenchmarking in the cloud. How bad is it really?"](https://link.springer.com/article/10.1007/s10664-019-09681-1) ([pdf](https://sci-hub.tw/10.1007/s10664-019-09681-1)), which uses the work of Tomas Kalibera and Richard Jones.

They find that, yeah you get some variability, but you can still get meaningful results in the cloud.

Reply:
> Looking at the diff.csv, it contains as many rows/datapoints as benchmark run. I was thinking more to a aggregated/summary output like the sentence you propose

Hm, when I run it the diff seams to do some rollup, it has only 20 lines, whereas the original has 182.

It should have the same number of lines as the lines of phrasing that are output.